src/main/java/com/redhat/devtools/intellij/quarkus/telemetry/TelemetryManager.java

```diff
@@ -0,0 +1,115 @@
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+/**
+ * Helper to initialize Telemetry. A Telemetry service based on <a href="https://github.com/redhat-developer/intellij-redhat-telemetry">intellij-redhat-telemetry</a>,
+ * will be instantiated if that plugin is installed, a No-Op Telemetry stub will be used otherwise.
+ */
+public class TelemetryManager implements Disposable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TelemetryManager.class);
+
+    private final Lazy<TelemetryMessageBuilder> builder = new Lazy<>(() -> new TelemetryMessageBuilder(PluginManager.getPluginByClass(this.getClass())));
+
+    private boolean hasError;
+
+    public static TelemetryManager instance() {
+        return ApplicationManager.getApplication().getService(TelemetryManager.class);
+    }
+
+    /**
+     * Sends a tracking event without additional properties.
+     * @param eventName the name of the event
+     */
+    public void send(@NotNull TelemetryEventName eventName) {
+        send(eventName, null);
+    }
+
+    /**
+     * Sends a tracking event with additional properties.
+     * @param eventName the name of the event
+     * @param properties the properties of the event
+     */
+    public void send(@NotNull TelemetryEventName eventName, @Nullable Map<String, String> properties) {
+        TelemetryMessageBuilder.ActionMessage action = action(eventName, properties);
+        if (action == null) {
+            return;
+        }
+        asyncSend(action);
+    }
+
+    @Nullable
+    public TelemetryMessageBuilder.ActionMessage action(@NotNull TelemetryEventName eventName) {
+        return action(eventName, null);
+    }
+
+    @Nullable
+    public TelemetryMessageBuilder.ActionMessage action(@NotNull TelemetryEventName eventName, @Nullable Map<String, String> properties) {
+        TelemetryMessageBuilder builder = getMessageBuilder();
+        if (builder == null) {
+            return null;
+        }
+        TelemetryMessageBuilder.ActionMessage action = builder.action(eventName.getEventName());
+        if (properties != null) {
+            properties.forEach((k, v) -> {
+                action.property(k, v);
+            });
+        }
+        return action;
+    }
+
+    @Nullable
+    private TelemetryMessageBuilder getMessageBuilder() {
+        if (hasError) {
+            return null;
+        }
+        try {
+            return builder.get();
+        }
+        catch(Exception e) {
+            hasError = true;
+            return null;
+        }
+    }
+
+    public void asyncSend(@Nullable TelemetryMessageBuilder.ActionMessage message) {
+        if (message == null) {
+            return;
+        }
+        ApplicationManager.getApplication().executeOnPooledThread(() -> {
+            try{
+                message.send();
+            } catch (Exception e) {
+                LOGGER.warn("Failed to send Telemetry data : {}", e.getMessage());
+            }
+        });
+    }
```

Comment:
I think we can keep the action and asyncSend methods private, no need to expose the upstream telemetry classes

Reply:
fixed